src/views/self/view_weapons.cpp

```diff
@@ -172,6 +173,16 @@ namespace big
 			ImGui::Checkbox("RAINBOW_PAINT"_T.data(), &g.weapons.paintgun.rainbow);
 			ImGui::SliderFloat("VIEW_WEAPON_RAINBOW_SPEED"_T.data(), &g.weapons.paintgun.speed, 0.f, 10.f);
 			if (!g.weapons.paintgun.rainbow) { ImGui::ColorEdit4("VIEW_WEAPON_PAINT_GUN_COLOR"_T.data(), g.weapons.paintgun.col); }
+			break;
+		case CustomWeapon::SHARK_GUN:
+			static char animal_gun[32];
+			std::memcpy(animal_gun, g.weapons.animal.c_str(), 32);
+			if (ImGui::InputTextWithHint("SHOOTING_MODEL"_T.data(), "NAME_ANIMAL_MODEL"_T.data(), animal_gun, sizeof(animal_gun)))
```

Comment:
Use components::input_text_with_hint instead of invoking the ImGui item directly. Also use IM_ARRAYSIZE on animal_gun as it is a hard requirement from ImGui.